src/app/components/batch-token-balance-adjustment-modal/batch-token-balance-adjustment-modal.component.ts

```diff
@@ -285,28 +260,27 @@ export class BatchTokenBalanceAdjustmentModalComponent {
         if (this.errorCollection == null) {
             throw new Error('Error collector has not been initialized');
         }
-        const noLinebreaks = errorMessage.replaceAll(/[\n\r]/g, '  ');
+        const error_noLinebreaks = errorMessage.replaceAll(/[\n\r]/g, '  ');
         if (Array.isArray(record)) {
-            this.errorCollection.push([...record, `line ${line}`, noLinebreaks]);
+            this.errorCollection.push([...record, `line ${line}`, error_noLinebreaks]);
         } else {
             this.errorCollection.push({
                 ...record,
                 Line: `${line}`,
-                Error: noLinebreaks
+                Error: error_noLinebreaks
             });
         }
     }
 
-    makeErrorCSV() {
+    async makeErrorCSV() {
         if (this.errorCollection == null || this.errorCollection.length == 0) return;
 
-        this.errorsCSVFile = new File(
-            [CSVParse.unparse(this.errorCollection)],
-            `${this.cleanName(this.selectedFile?.name)}-Errors-${this.cleanName(new Date().toISOString())}.csv`,
-            { type: 'text/csv;charset=utf-8;' }
-        );
+        const importedFileName = this.selectedFile?.name == null ? 'Batch-CSV' : this.selectedFile?.name;
```

Comment:
[The nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) could be used here.